realm/realm-jni/src/io_realm_internal_SharedGroup.cpp

```diff
@@ -323,4 +323,25 @@ JNIEXPORT jlongArray JNICALL Java_io_realm_internal_SharedGroup_nativeGetVersion
     env->SetLongArrayRegion(version_data, 0, 2, version_array);
 
     return version_data;
-}
\ No newline at end of file
+}
+
+JNIEXPORT jboolean JNICALL Java_io_realm_internal_SharedGroup_nativeWaitForChange
+  (JNIEnv *, jobject, jlong native_ptr)
+{
+    TR_ENTER_PTR(native_ptr)
+    return SG(native_ptr)->wait_for_change();
```

Comment:
`wait_for_change()` might throw exceptions, and `try`/`catch` might be appropriate.